projects/libaom/av1_dec_fuzzer.cc

```diff
@@ -0,0 +1,91 @@
+// Fuzzing of AV1 decoder.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "aom/aom_decoder.h"
+#include "aom/aomdx.h"
+#include "aom_ports/mem_ops.h"
+#include "aom/common/ivfdec.h"
+
+// TODO(urvang): Create a 2nd fuzz target that defines DECODE_MODE_threaded.
+#define DECODE_MODE_serial
+
+static const char *const kIVFSignature = "DKIF";
+
+extern "C" void usage_exit(void) { exit(EXIT_FAILURE); }
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  FILE *const file = fmemopen((void *)data, size, "rb");
+  if (file == nullptr) {
+    fprintf(stderr, "Failed to get file stream.\n");
+    return 0;
+  }
+
+  char header[32];
+  if (fread(header, 1, 32, file) != 32) {
```

Comment:
Do these three if statements and their bodies test libaom code? Are they necessary for fuzzing (ie: will ivf_read_frame crash if file is less than 32 bytes)? 
If not, it makes sense to remove them since they are just slowing things down (marginally).

Reply:
The first fread() is required to read the file header, yes.
[ivf_read_frame() assumes that file header is already read, and reads the subsequent frame bytes].
But yes, others were not really required -- removed.